manifests/params.pp

```diff
@@ -2,69 +2,204 @@
+      $print_motd        = true
+      $permit_root_login = false
+      $dos_line_endings  = false
     }
     'CumulusLinux', 'Debian', 'Ubuntu': {
-      $client_package  = 'openssh-client'
-      $server_package  = 'openssh-server'
-      $ssh_service     = 'ssh'
-      $syslog_facility = 'AUTHPRIV'
-      $print_motd      = false
+      $server_class      = 'ssh::server::linux'
+      $client_package    = 'openssh-client'
+      $server_package    = 'openssh-server'
+      $package_provider  = undef # Use default
+      $ssh_service       = 'ssh'
+      $has_restart       = true
+      $config_dir        = '/etc/ssh'
+      $config_owner      = 'root'
+      $config_group      = '0'
+      $config_mode       = '0644'
+      $config_dir_mode   = '0755'
+      $root_access_group = 'root'
+      $manage_config_dir = true
+      $syslog_facility   = 'AUTHPRIV'
+      $sftp_subsystem    = 'internal-sftp'
+      $authorized_keys   = '.ssh/authorized_keys'
+      $strict_modes      = true
+      $print_motd        = false
+      $permit_root_login = false
+      $dos_line_endings  = false
     }
     'Darwin': {
-      $client_package  = undef
-      $server_package  = undef
-      $ssh_service     = 'com.openssh.sshd'
-      $syslog_facility = 'AUTHPRIV'
-      $print_motd      = true
+      $server_class      = undef
+      $client_package    = undef
+      $server_package    = undef
+      $package_provider  = undef # Use default
+      $ssh_service       = 'com.openssh.sshd'
+      $has_restart       = false
+      $config_dir        = '/etc/ssh'
+      $config_owner      = 'root'
+      $config_group      = '0'
+      $config_mode       = '0644'
+      $config_dir_mode   = '0755'
+      $root_access_group = 'wheel'
+      $manage_config_dir = true
+      $syslog_facility   = 'AUTHPRIV'
+      $sftp_subsystem    = 'internal-sftp'
+      $authorized_keys   = '.ssh/authorized_keys'
+      $strict_modes      = true
+      $print_motd        = true
+      $permit_root_login = false
+      $dos_line_endings  = false
     }
     'FreeBSD': {
-      $ssh_service     = 'sshd'
-      $syslog_facility = 'AUTHPRIV'
-      $print_motd      = true
+      $server_class      = undef
+      $package_provider  = undef # Use default
+      $client_package    = 'openssh-portable'
+      $server_package    = 'openssh-portable'
+      $ssh_service       = 'sshd'
+      $has_restart       = true
+      $config_dir        = '/etc/ssh'
+      $config_owner      = 'root'
+      $config_group      = '0'
+      $config_mode       = '0644'
+      $config_dir_mode   = '0755'
+      $root_access_group = 'wheel'
+      $manage_config_dir = true
+      $syslog_facility   = 'AUTHPRIV'
+      $sftp_subsystem    = 'internal-sftp'
+      $authorized_keys   = '.ssh/authorized_keys'
+      $strict_modes      = true
+      $print_motd        = true
+      $permit_root_login = false
+      $dos_line_endings  = false
     }
     'Solaris','SunOS': {
       case $::kernelrelease {
```

Comment:
Mind updating to the current structured fact here?

Reply:
Good call.